api/v1beta2/imagerepository_types.go

```diff
@@ -116,6 +116,11 @@ type ImageRepositorySpec struct {
 }
 
 type ScanResult struct {
+	// Revision is a stable hash of the scanned tags.
+	// TODO: Remove optional in API v1.
+	// +optional
+	Revision string `json:"revision"`
+
 	TagCount   int         `json:"tagCount"`
 	ScanTime   metav1.Time `json:"scanTime,omitempty"`
 	LatestTags []string    `json:"latestTags,omitempty"`
```

Comment:
Can you please add comments to each field and set `+optional`.  Missed this when we introduced  `ScanResult`. Also please add a comment to the `ScanResult` type.

Reply:
The only place `.status.lastScanResult` receives a value, it's here:

```go
	scanTime := metav1.Now()
	obj.Status.LastScanResult = &imagev1.ScanResult{
		Revision:   checksum,
		TagCount:   len(filteredTags),
		ScanTime:   scanTime,
		LatestTags: latestTags,
	}
```

Are these fields really optional? They're always set :thinking: 

`.status.lastScanResult` is already optional itself:

```go
	// LastScanResult contains the number of fetched tags.
	// +optional
	LastScanResult *ScanResult `json:"lastScanResult,omitempty"`
```

Reply:
I guess latestTags can be empty, what if the repo has no images? But all others should be required, we'll change them in v1.